### ISSUE TITLE
when a node is both stretch and auto, get the result with width and height "null"

### ISSUE THREAD

**Issue by Lizz0y** [facebook/yoga#644]
# Report

when a node is both cross stretch and margin auto, get the result with width and height "null",i have read the source code and found the following code:

```
          if (alignItem == YGAlignStretch &&
              YGMarginLeadingValue(child, crossAxis)->unit != YGUnitAuto &&
              YGMarginTrailingValue(child, crossAxis)->unit != YGUnitAuto) {
```

can someone explain , since the alignItem is already stretch ,auto is meaningless, so why this code exclude the auto ?

**Comment by woehrl01** [comment]
It excludes auto, as if you have a margin of auto. The node should not stretch. Margin is "stronger" than align-item.

**Comment by Lizz0y** [comment]
i write a node with "text" type ,since text has no width&height,so i set a custom measureFunction as:

```
 YogaNode root = new YogaNode();
 root.setWidth(500);
 root.setHeight(300);

        YogaNode text = new YogaNode();
        text.setMeasureFunction(new YogaMeasureFunction() {
            @Override
            public long measure(YogaNode node, float width, YogaMeasureMode widthMode, float height, YogaMeasureMode heightMode) {
                return new YogaMeasureOutput().make(50,50);
            }
        });
       text.setMarginAuto(YogaEdge.LEFT);
        root.addChildAt(text, 0);

```

when i run this code ,find the text node return "null" for both height and width, please find what's wrong with it ,thx

**Comment by woehrl01** [comment]
It should be a width and height of 50 and left 450 here. I'll reproduce and have a look.

**Comment by woehrl01** [comment]
thank you for the hint @Lizz0y , I pushed a fix for this in #645

**Comment by Lizz0y** [comment]
thx @woehrl01  for fix this problem
